tests/src/test/scala/org/http4s/parser/Rfc3986ParserSpec.scala

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2013-2020 http4s.org
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.http4s.parser
+
+import org.http4s.Http4sSpec
+import org.http4s.Uri.Ipv6Address
+import org.http4s.internal.parsing.Rfc3986
+import org.scalacheck.Prop
+
+class Rfc3986ParserSpec extends Http4sSpec {
+  "Rfc3986 parser" should {
+    "parse ipv6" in {
+      Prop.forAll(http4sTestingArbitraryForIpv6Address.arbitrary) { (ipv6: Ipv6Address) =>
```

Comment:
It's great to have a property test, but as you work through, add one example of each shape at a time.  Keep it green, and build it back up!